src/app/data/data.component.html

```diff
@@ -14,7 +14,7 @@
     <button
       (click)="submitData()"
       class="btn btn-primary"
-      [disabled]="!selectedArea"
+      [disabled]="!selectedArea || springsLoading"
```

Comment:
Here we disable the button in 2 cases:

1. Value from the dropdown is not selected
2. Request is sent, and button disabled until it is finished

To only disable button when request is running:

```js
[disabled]="springsLoading"
```